packages/devtools/kit/src/channel.ts

```diff
@@ -0,0 +1,156 @@
+import { nanoid } from 'nanoid';
+import type { ChannelOptions } from 'birpc';
+
+export class WebSocketChannel implements ChannelOptions {
+  static link(ws: WebSocket) {
+    return new Promise<WebSocketChannel>((resolve, reject) => {
+      const ret = new WebSocketChannel(ws);
+      if (ws.readyState === ws.OPEN) {
+        resolve(ret);
+      } else {
+        ws.addEventListener('open', () => resolve(ret), { once: true });
+        ws.addEventListener('error', reject, { once: true });
+        ws.addEventListener('close', reject, { once: true });
+        setTimeout(reject, 1000);
+      }
+    });
+  }
+
+  protected ws: WebSocket;
+
+  constructor(ws: WebSocket) {
+    this.ws = ws;
+    const binds: ChannelOptions = {
+      on: this.on.bind(this),
+      post: this.post.bind(this),
+      deserialize: this.deserialize.bind(this),
+      serialize: this.serialize.bind(this),
+    };
+    Object.assign(this, binds);
+  }
+
+  get handlers() {
+    return {
+      on: this.on.bind(this),
+      post: this.post.bind(this),
+      deserialize: this.deserialize.bind(this),
+      serialize: this.serialize.bind(this),
+    };
+  }
+
+  post(data: any, ...extras: any[]): any {
+    if (extras.length) {
+      throw new Error('Unexpected extra parameters.');
+    }
+    this.ws.send(data);
+  }
+
+  on(fn: (data: any, ...extras: any[]) => void): any {
+    this.ws.onmessage = fn;
+  }
+
+  serialize(data: any): any {
+    return JSON.stringify(data);
+  }
+
+  deserialize(e: any): any {
+    return JSON.parse(e.data.toString());
+  }
+}
+
+export class MessagePortTimeout extends Error {
+  constructor() {
+    super('Connection to target message channel port timed out');
+  }
+}
+
+export type PostMessageListener = (ev: MessageEvent) => void;
+
+export interface PostMessageTarget {
+  postMessage: (
+    message: any,
```

Comment:
can we use unknown instead of any

Reply:
so many anys in this PR...

Reply:
It just implement the [`ChannelOptions`](https://github.com/antfu/birpc/blob/b82bef7dbc804525be7b4afca127997160cf9830/src/index.ts#L9) interface from `birpc` and won't be used by any other developers but just pass to the birpc instance. But I found the restrict typing won't make it unaccpetable to birpc. So I did it in the latest commit.